### PR TITLE
XSOverlay: Return old API for compatibility 

### PR DESCRIPTION
**Pull request by nyakowint**
beats the allegations i mean resolves this oversight https://github.com/Vendicated/Vencord/pull/2736#issuecomment-2261782256 (?)

### CODE REVIEW

File: src/plugins/xsOverlay/index.tsx [hunk 1]
```diff
@@ -349,6 +356,10 @@ function sendOtherNotif(content: string, titleString: string) {
 }
 
 async function sendToOverlay(notif: NotificationObject) {
+    if (!IS_WEB && settings.store.preferUDP) {
```
Comment (Sqaaakoi) [inline src/plugins/xsOverlay/index.tsx:359]: Vesktop can use plugin natives iirc

File: src/plugins/xsOverlay/index.tsx [hunk 2]
```diff
@@ -102,6 +102,11 @@ const settings = definePluginSettings({
             await start();
         }
     },
+    preferUDP: {
+        type: OptionType.BOOLEAN,
+        description: "Enable if you use an older build of XSOverlay unable to connect through websockets. This setting is ignored on web.",
+        default: false
```
Comment (Vendicated) [inline src/plugins/xsOverlay/index.tsx:108]:
```suggestion
        default: false,
        disabled: () => IS_WEB
```

File: src/plugins/xsOverlay/index.tsx [hunk 3]
```diff
@@ -178,6 +183,8 @@ async function start() {
     });
 }
 
+const Native = VencordNative.pluginHelpers.XsOverlay as PluginNative<typeof import("./native")>;
```
Comment (Vendicated) [inline src/plugins/xsOverlay/index.tsx:186]: do u even test 😭 
```suggestion
const Native = VencordNative.pluginHelpers.XSOverlay as PluginNative<typeof import("./native")>;
```

File: src/plugins/xsOverlay/index.tsx (hunk 2 above)
Comment (nyakowint) [reply src/plugins/xsOverlay/index.tsx:108]: I didnt know you could do that lol

File: src/plugins/xsOverlay/index.tsx (hunk 1 above)
Comment (nyakowint) [reply src/plugins/xsOverlay/index.tsx:359]: actual web cant lol

**Pull request merged**